pennylane/devices/qutrit_mixed/measure.py

```diff
@@ -0,0 +1,296 @@
+    """Measure the expectation value of an observable by finding the trace of obs@rho.
+
+    Args:
+        measurementprocess (ExpectationMP): measurement process to apply to the state.
+        state (TensorLike): the state to measure.
+        is_state_batched (bool): whether the state is batched or not.
+
+    Returns:
+        TensorLike: the result of the measurement.
+    """
+    obs = measurementprocess.obs
+    rho_mult_obs = apply_observable_einsum(obs, state, is_state_batched)
+
+    # using einsum since trace function axis selection parameter names
+    # are not consistent across interfaces, they don't exist for torch
+
+    num_wires = get_num_wires(state, is_state_batched)
+    rho_mult_obs_reshaped = reshape_state_as_matrix(rho_mult_obs, num_wires)
+    if is_state_batched:
+        return math.real(math.stack([math.sum(math.diagonal(dm)) for dm in rho_mult_obs_reshaped]))
+
+    return math.real(math.sum(math.diagonal(rho_mult_obs_reshaped)))
+
+
+def calculate_reduced_density_matrix(  # TODO: ask if I should have state diagonalization gates?
+    measurementprocess: StateMeasurement, state: TensorLike, is_state_batched: bool = False
+) -> TensorLike:
+    """Get the state or reduced density matrix.
+
+    Args:
+        measurementprocess (StateMeasurement): measurement to apply to the state.
+        state (TensorLike): state to apply the measurement to.
+        is_state_batched (bool): whether the state is batched or not.
+
+    Returns:
+        TensorLike: the resulting resquared density_matrix.
+    """
+    wires = measurementprocess.wires
+    if not wires:
+        return reshape_state_as_matrix(state, get_num_wires(state, is_state_batched))
+
+    num_obs_wires = len(wires)
+    num_state_wires = get_num_wires(state, is_state_batched)
+    state_wire_indices_list = list(alphabet[:num_state_wires] * 2)
+    final_state_wire_indices_list = [""] * (2 * num_obs_wires)
+
+    for i, wire in enumerate(wires):
+        col_index = wire + num_state_wires
+        state_wire_indices_list[col_index] = alphabet[col_index]
+        final_state_wire_indices_list[i] = alphabet[wire]
+        final_state_wire_indices_list[i + num_obs_wires] = alphabet[col_index]
+
+    state_wire_indices = "".join(state_wire_indices_list)
+    final_state_wire_indices = "".join(final_state_wire_indices_list)
+
+    state = math.einsum(f"...{state_wire_indices}->...{final_state_wire_indices}", state)
+
+    return reshape_state_as_matrix(state, len(wires))
+
+
+def calculate_probability(
+    measurementprocess: StateMeasurement, state: TensorLike, is_state_batched: bool = False
+) -> TensorLike:
+    """Find the probability of measuring states.
+
+    Args:
+        measurementprocess (StateMeasurement): measurement to apply to the state.
+        state (TensorLike): state to apply the measurement to.
+        is_state_batched (bool): whether the state is batched or not.
+
+    Returns:
+        TensorLike: the result of the measurement.
```

Comment:
Please make these more descriptive and tailored to the function (comment applies to other functions too)